schema.graphql

```diff
@@ -343,14 +343,34 @@ type MinterFilter @entity {
   "Unique identifier made up of minter filter contract address"
   id: ID!
 
-  "Associated core contract"
-  coreContract: Contract!
+  "Minters allowlisted globally on this MinterFilter"
+  minterGlobalAllowlist: [Minter!]!
 
-  "Minters allowlisted on MinterFilter"
-  minterAllowlist: [Minter!]!
+  "Minters allowlisted at a contract-level on this MinterFilter, only used for MinterFilterV2+"
+  minterFilterContractAllowlists: [MinterFilterContractAllowlist!]!
+    @derivedFrom(field: "minterFilter")
 
-  "Minters associated with MinterFilter"
-  associatedMinters: [Minter!]! @derivedFrom(field: "minterFilter")
+  "Known minters that are tied to this MinterFilter, but are not necessarily approved on this MinterFilter"
+  knownMinters: [Minter!]! @derivedFrom(field: "minterFilter")
+
+  "Core contract registry used by this MinterFilter. Note that for MinterFilterV0 & MinterFilterV1, a dummy CoreRegistry is created in the subgraph that contains a single allowlisted core contract."
+  coreRegistry: CoreRegistry!
+
+  updatedAt: BigInt!
+}
+
+type MinterFilterContractAllowlist @entity {
+  "Unique identifier made up of {minter filter contract address}-{core contract address}"
+  id: ID!
+
+  "MinterFilter contract"
+  minterFilter: MinterFilter!
+
+  "Core contract"
+  contract: Contract!
+
+  "Minter contract addresses allowed at the contract level (extending global MinterFilter allowlist)"
+  minterContractAllowlist: [Bytes!]!
```

Comment:
Is there a reason this is a `Bytes` array while the global allowlist is a `Minter` array?

Is this because we expect engine partners to use minters that we haven't seen before / aren't indexing ourselves, but should still be allowed to mint?

Reply:
good catch - I think this actually should be an array of `[Minter!]!`, so I've update.
We do expect engine partners to use minters that we haven't seen before/aren't indexing, but I think we would still generically be fine creating Minter entities in the store from them, and therefore should have an array of `Minter`s here.

Do want to note that an array here is certainly a design choice, and we expect this specific array to remain relatively short. We could make this a "derived from" field if we added yet another entity `MinterFilterContractAllowlisting`, that defined each individual minter-core-minterfilter allowlisting...

Reply:
resolving this since it has been updated, and the design choice noted in the internal design doc.